sway-core/src/parse_tree/expression/mod.rs

```diff
@@ -489,31 +517,38 @@ impl Expression {
                 }
                 // this is non-optional and part of the parse rule so it won't fail
                 let name = name.unwrap();
-                Expression::VariableExpression { name, span }
+                let exp = Expression::VariableExpression { name, span };
+                ParseResult {
+                    var_decls: vec![],
+                    value: exp,
+                }
```

Comment:
I see this pattern a lot, what about a parse result method that looks like `ParseResult::empty(exp)` that makes one with no var decls?